lib/features/parkings_view/repository/parkings_repository.dart

```diff
@@ -26,6 +27,26 @@ Future<IList<Parking>> parkingsRepository(ParkingsRepositoryRef ref) async {
   final response = await ref.postIParkingCommand<Map<String, dynamic>>(
     FetchPlacesCommand(DateTime.now()),
   );
-  final list = response.data?["places"] as List<dynamic>;
-  return list.whereType<Map<String, dynamic>>().map(Parking.fromJson).toIList();
+  final parkings = response.data?["places"] as List<dynamic>;
+  final List<Parking> list =
+      parkings.whereType<Map<String, dynamic>>().map(Parking.fromJson).toList();
+
+  return _sortParkingsByFav(list, ref).toIList();
+}
+
+List<Parking> _sortParkingsByFav(
+  List<Parking> list,
+  ParkingsRepositoryRef ref,
+) {
+  final List<Parking> finalParkings = [];
+  for (final parking in list) {
+    final isFavorite =
+        ref.watch(localFavParkingsRepositoryProvider(parking.id)) ?? false;
+    if (isFavorite) {
+      finalParkings.insert(0, parking);
```

Comment:
I know this talk has 0 sense when we have only 5 parkings and pwr probably won't build next 1000 or 10k in a night - but when I see inserting elements at 0 index of an ArrayList in a loop, I do see this as a redflag :)

On a big datasets this will be pretty heavy operation - maybe you remember this from java classes from uni (xd). I assume you more or less know what I mean, but if you want deeper explanation, I'm more than happy to explain this.

Default Dart's list is some sort of array list implementation.

And you only insert at the beggining and the end and then right away use `toIList` conversion, which iterates over collection 1 time to create a completely different data structure - so this list doesn't need random index access.

IMO this seems as a perfect occasion to use some sort of Linked List implementation, e.g. `DoubleLinkedQueue` from `collections` package. We already use this package and this should be a pretty quick change. 

But for this size of dataset your solution is more than valid, so you can leave it too if you want it.